src/java.base/share/classes/java/util/stream/Gatherer.java

```diff
@@ -0,0 +1,566 @@
+     *         result(s) which are then passed on to the provided Downstream
+     */
+    default BiConsumer<A, Downstream<? super R>> finisher() {
+        return defaultFinisher();
+    }
+
+    /**
+     * Returns a composed Gatherer which connects the output of this Gatherer
+     * to the input of that Gatherer.
+     *
+     * @param that the other gatherer
+     * @param <AA> The type of the state of that Gatherer
+     * @param <RR> The type of output of that Gatherer
+     * @throws NullPointerException if the argument is null
+     * @return returns a composed Gatherer which connects the output of this
+     *         Gatherer as input that Gatherer
+     */
+    default <AA, RR> Gatherer<T, ?, RR> andThen(Gatherer<? super R, AA, ? extends RR> that) {
+        Objects.requireNonNull(that);
+        return Gatherers.Composite.of(this, that);
+    }
+
+    /**
+     * Returns an initializer which is the default initializer of a Gatherer.
+     * The returned initializer identifies that the owner Gatherer is stateless.
+     *
+     * @see Gatherer#initializer()
+     * @return the instance of the default initializer
+     * @param <A> the type of the state of the returned initializer
+     */
+    static <A> Supplier<A> defaultInitializer() {
+        return Gatherers.Value.DEFAULT.initializer();
+    }
+
+    /**
+     * Returns a combiner which is the default combiner of a Gatherer.
+     * The returned combiner identifies that the owning Gatherer must only
+     * be evaluated sequentially.
+     *
+     * @see Gatherer#finisher()
+     * @return the instance of the default combiner
+     * @param <A> the type of the state of the returned combiner
+     */
+    static <A> BinaryOperator<A> defaultCombiner() {
+        return Gatherers.Value.DEFAULT.combiner();
+    }
+
+    /**
+     * Returns a {@code finisher} which is the default finisher of
+     * a {@code Gatherer}.
+     * The returned finisher identifies that the owning Gatherer performs
+     * no additional actions at the end of input.
+     *
+     * @see Gatherer#finisher()
+     * @return the instance of the default finisher
+     * @param <A> the type of the state of the returned finisher
+     * @param <R> the type of the Downstream of the returned finisher
+     */
+    static <A,R> BiConsumer<A, Downstream<? super R>> defaultFinisher() {
+        return Gatherers.Value.DEFAULT.finisher();
+    }
+
+    /**
+     * Returns a new, sequential, and stateless {@code Gatherer} described by
+     * the given {@code integrator}.
+     *
+     * @param integrator the integrator function for the new gatherer
+     * @param <T> the type of input elements for the new gatherer
+     * @param <R> the type of results for the new gatherer
+     * @throws NullPointerException if the argument is null
+     * @return the new {@code Gatherer}
+     */
+    static <T, R> Gatherer<T, Void, R> ofSequential(
+            Integrator<Void, T, R> integrator) {
```

Comment:
I've thought about that, and since we don't use that for the other parameters (like `initializer`) I don't think it helps changing it here. And the reason this has little effect is that the code calling the factory typically controls the components/parameters/return type.

Reply:
It's still possible to have a situation where PECS signature could be useful, and I don't see any downsides. A user may want to reuse the same static integrator instead of creating several identical lambdas just because the target type is more specific. Sometimes, people complain later, and you'll have to fix this. For example, see that `Stream.generate()` [was fixed separately](https://bugs.openjdk.org/browse/JDK-8132097). So why not doing this from the very beginning?

Reply:
@amaembo In this case, given that it is intended as Preview, I think it makes sense to receive some user feedback on this. Note that Collector has been without PECS for Collector.of since 1.8: [of](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collector.html#of-java.util.function.Supplier-java.util.function.BiConsumer-java.util.function.BinaryOperator-java.util.function.Function-java.util.stream.Collector.Characteristics...-)

Right now I don't want to deviate from the Collector-heritage until there's more usage feedback, and I'm definitely not ruling out that PECS:ing the factories might be worth it.

Reply:
I don't see the point of waiting here. Usually requests for PECSing signatures always comes two or three years after the API is realing when people starts to use the API in anger so the preview time is likely not enough.

Reply:
Postponing such a change as candidate for a potential second preview.